### PR TITLE
persist: storage client fn to get total usage in one go

### PR DESCRIPTION
**Pull request by pH14**
I think this should make @jpepin's PR easier -- we can total size for all blobs by shard id in one go (and unattributable storage usage, if any, hopefully not), and then the caller can correlate those shard ids to their sources/tables afterwards.

### Motivation

 * This PR refactors existing code.

https://materializeinc.slack.com/archives/C03NYM894GP/p1659042453673699

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-protobuf` label.
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - N/A

### CODE REVIEW

File: src/persist-client/src/usage.rs [hunk 1]
```diff
@@ -44,6 +45,35 @@ impl StorageUsageClient {
         .await
     }
 
+    /// Returns a map of [crate::ShardId] to its size (in bytes) stored in persist,
+    /// and the total bytes stored for unattributable data (i.e. does not have a shard
+    /// key in its name). This latter count _should_ always be 0, so being nonzero
+    /// indicates either persist wrote an invalid blob key, or another process is
+    /// storing data under the same path (!)
+    pub async fn shard_sizes(&self) -> (HashMap<ShardId, u64>, u64) {
```
Comment (danhhz) [inline src/persist-client/src/usage.rs:53]: it feels more natural to me for the map key to be `Option<ShardId>` and then use `None` for the unattributable data, but happy to defer to you or @jpepin

File: src/persist-client/src/usage.rs [hunk 2]
```diff
@@ -140,5 +170,17 @@ mod tests {
 
         assert_eq!(usage.shard_size(&shard_id_one).await, shard_one_size);
         assert_eq!(usage.shard_size(&shard_id_two).await, shard_two_size);
+
+        let (shard_sizes, unattributable) = usage.shard_sizes().await;
+        assert_eq!(unattributable, 0);
+        assert_eq!(shard_sizes.len(), 2);
+        assert_eq!(
+            *shard_sizes.get(&shard_id_one).expect("must have shard 1"),
```
Comment (danhhz) [inline src/persist-client/src/usage.rs:178]: nit: instead of the * and the expect, I'd assert_eq against `Some(&shard_one_size)`

**Review by danhhz — APPROVED** [review]
LGTM!

File: src/persist-client/src/usage.rs (hunk 2 above)
Comment (pH14) [reply src/persist-client/src/usage.rs]: oh yeah that reads a lot better. I feel like I don't fully understand references to Copy types in Rust

**Pull request merged**